http/servlet-undertow/src/test/java/io/quarkus/ts/http/undertow/HttpServletWithSessionListenerIT.java

```diff
@@ -90,4 +140,9 @@ private void thenWaitToEvictSessionsAndCheckActiveSessionsEqualTo(int value) {
                 .pollInterval(REST_ASSURANCE_POLL_INTERVAL)
                 .until(() -> getActiveSessions() == value);
     }
+
+    @AfterAll
+    public static void closeHttpClientConnection() {
+        httpClient.close();
```

Comment:
FYI - closing `httpClient` that is only created in one of 3 test methods without null check is wrong because during testing you can just run one of the test methods (e.g. when debugging stuff)

Reply:
I understand your point but I think in this test class context the `httpClient` variable is static, so it persists across the execution of the test class,so in this context using `@AfterAll` and closing the `httpclient` connection is still valid, and you can execute the other tests independently and it won't produce any failure...

Reply:
you can't execute other tests independently without NPE, just try to run `sessionEviction`

Reply:
![Screenshot from 2024-03-01 17-38-20](https://github.com/quarkus-qe/quarkus-test-suite/assets/75727563/1fadbdcb-1a6e-495b-8ad1-7227a8419ae4)

![image](https://github.com/quarkus-qe/quarkus-test-suite/assets/75727563/8e71684a-f596-4522-9e54-db74975d0a7d)

Reply:
Executed independently through IDE(IntelliJ) and with mvn command ` servlet-undertow]$ mvn clean verify -Dit.test=HttpServletWithSessionListenerIT#sessionSecured
` Both passed

Reply:
Thank you, I didn't know. NPE is thrown ans swallowed https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/LifecycleMethodExecutionExceptionHandler.html#implementation-guidelines. This NPE would only have negative effect if there were other `@AfterAll` (multiple) methods, which there is not. So it okay-ish (I still don't like it...). Thanks